docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -276,23 +355,101 @@ serializationConfig.
         .register(Foo.class, "foo", new FooSerializer()); // Use the "foo" as the type name
 ----
 
+**Declarative Configuration:**
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    ...
+    <serialization>
+        <compact-serialization enabled="true">
+            <registered-classes>
+                <class type-name="foo" serializer="com.example.FooSerializer">
+                    com.example.Foo
+                </class>
+            </registered-classes>
+        </compact-serialization>
+    </serialization>
+    ...
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  serialization:
+    compact-serialization:
+      enabled: true
+      registered-classes:
+        - class: com.example.Foo
+          type-name: foo
+          serializer: com.example.FooSerializer
+----
+====
+
 Lastly, the following is a sample configuration that registers reflective
 serializer for a certain class, without implementing an explicit serializer.
 
+**Programmatic Configuration:**
+
 [source,java]
 ----
 SerializationConfig serializationConfig = new SerializationConfig();
 serializationConfig.
         getCompactSerializationConfig()
         .setEnabled(true)
-        .register(Bar.class); // Use the fully qualified class name as the type name
+        .register(Bar.class); // Uses the fully qualified class name as the type name
+----
+
+**Declarative Configuration:**
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    ...
+    <serialization>
+        <compact-serialization enabled="true">
+            <registered-classes>
+                <class>com.example.Bar</class>
+            </registered-classes>
+        </compact-serialization>
+    </serialization>
+    ...
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  serialization:
+    compact-serialization:
+      enabled: true
+      registered-classes:
+        - class: com.example.Bar
 ----
+====
 
 You may need to register a class for reflective serializer if 
```

Comment:
do you mean a case where user class is for example implementing `IdentifiedDataSerializable` but still want 't serialized as compact? If this is the aim, we should say that we do not recommend such uses. Also, then you should explicitly tell this case, and should also tell that it should not be registered to other factory classes, etc.

Reply:
@sancar wrote that part so he may answer the best https://github.com/hazelcast/hz-docs/pull/120
but, as far as I see, it is just saying that if you have a class that is meant to be serialized with other mechanisms, and you want to use compact for it, you can register a reflective serializer and it will take a priority over it

Reply:
@ihsandemir I have added this document specifically for Serializable but to make it complete I have added all the other methods. I haven't thought about what it would mean for our methods. I agree, we should make it clearer as you suggested.

Reply:
@ihsandemir @sancar See https://github.com/hazelcast/hz-docs/pull/122/commits/8471e2036014fe2cd50249c72d274a2b05febbe9